File: services/analytics-service/app/api/v1/endpoints/analytics.py
from uuid import UUID
from fastapi import APIRouter, Depends, Query
from sqlalchemy.ext.asyncio import AsyncSession

from app.schemas.analytics import (
    DashboardSummary, FunnelResponse, ResumeComparisonResponse,
    ScoreCallbackResponse, TrendResponse,
)
from app.services.analytics_service import AnalyticsService
from resumeradar_common.auth.dependencies import get_current_user
from resumeradar_common.database.session import get_db

router = APIRouter()


@router.get("/analytics/dashboard", response_model=DashboardSummary)
async def get_dashboard(
    user_id: UUID = Depends(get_current_user), db: AsyncSession = Depends(get_db),
):
    service = AnalyticsService(db)
    return await service.get_dashboard_summary(user_id)


@router.get("/analytics/funnel", response_model=FunnelResponse)
async def get_funnel(
    user_id: UUID = Depends(get_current_user), db: AsyncSession = Depends(get_db),
):
    service = AnalyticsService(db)
    return await service.get_funnel(user_id)


@router.get("/analytics/resumes", response_model=ResumeComparisonResponse)
async def get_resume_comparison(
    user_id: UUID = Depends(get_current_user), db: AsyncSession = Depends(get_db),
):
    service = AnalyticsService(db)
    return await service.get_resume_comparison(user_id)


@router.get("/analytics/score-callback", response_model=ScoreCallbackResponse)
async def get_score_callback(
    user_id: UUID = Depends(get_current_user), db: AsyncSession = Depends(get_db),
):
    service = AnalyticsService(db)
    return await service.get_score_vs_callback(user_id)


@router.get("/analytics/trends", response_model=TrendResponse)
async def get_trends(
    period: str = Query(default="weekly", regex="^(weekly|monthly)$"),
    user_id: UUID = Depends(get_current_user), db: AsyncSession = Depends(get_db),
):
    service = AnalyticsService(db)
    return await service.get_trends(user_id, period=period)

File: frontend/src/api/analytics.ts
import { analyticsApi } from './client'
import type { DashboardSummary, FunnelResponse } from '../types/api'

export const analytics = {
  dashboard: async (): Promise<DashboardSummary> => {
    const { data } = await analyticsApi.get('/api/v1/analytics/dashboard')
    return data
  },

  funnel: async (): Promise<FunnelResponse> => {
    const { data } = await analyticsApi.get('/api/v1/analytics/funnel')
    return data
  },

  resumeComparison: async () => {
    const { data } = await analyticsApi.get('/api/v1/analytics/resumes')
    return data
  },

  scoreCallback: async () => {
    const { data } = await analyticsApi.get('/api/v1/analytics/score-callback')
    return data
  },

  trends: async (period: 'weekly' | 'monthly' = 'weekly') => {
    const { data } = await analyticsApi.get('/api/v1/analytics/trends', { params: { period } })
    return data
  },
}